### PR TITLE
Test RandomEcPoint hint & Fix for LinearSplit

### LINKED ISSUE #1115: Make a Cairo 1 program for testing RandomEcPoint hint
Plus add a unit test

### PR DESCRIPTION
**Pull request by rcatalan98**
# Test `RandomEcPoint` hint & Fix for LinearSplit

## Description

This PR adds a fix for LinearSplit and test support for `RandomEcPoint`

## Checklist
- [x] Linked to Github Issue
- [x] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

Closes #1115

### CODE REVIEW

**Comment by rcatalan98** [comment]
This PR is blocked until the new Hints are merged.

File: cairo_programs/cairo-1-contracts/random_ec_point.cairo
Comment (fmoletta) [inline cairo_programs/cairo-1-contracts/random_ec_point.cairo:11]:
```suggestion
}

```

**Comment by fmoletta** [comment]
Maybe we could also call ec_state_add and ec_state_finalize and check the result of the operation in the test contract

File: CHANGELOG.md
Comment (fmoletta) [inline CHANGELOG.md:6]: I wouldn't include this in the Changelog. As the code that was fixed was added in the same release the fix is "invisible" to those updating cairo-rs to a newer version.

**Review by Oppen — APPROVED** [review]

**Review by fmoletta — APPROVED** [review]
LGTM

**Pull request merged**
